opal/mca/accelerator/accelerator.h

```diff
@@ -0,0 +1,475 @@
+    void *ptr);
+
+/**
+ * Retrieves the base address and/or size of a memory allocation of the
+ * device.
+ *
+ * @param[IN] ptr            Pointer to device memory to get base/size from
+ * @param[OUT] base          Base address of the memory allocation
+ * @param[OUT] size          Size of the memory allocation
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_get_address_range_fn_t)(
+    const void *ptr, void **base, size_t *size);
+
+/**
+ * Queries whether the device supports Inter Process Communication
+ * or not. If true, the functions:
+ *
+ * opal_accelerator_base_module_get_handle_size_fn_t()
+ * opal_accelerator_base_module_get_handle_fn_t()
+ * opal_accelerator_base_module_open_handle_fn_t()
+ * opal_accelerator_base_module_close_handle_fn_t()
+ *
+ * must be implemented.
+ *
+ * @return true              IPC supported
+ * @return false             IPC not supported
+ */
+typedef bool (*opal_accelerator_base_module_is_ipc_enabled_fn_t)(void);
+
+/**
+ * Returns the size of the Inter Process Communication memory handle
+ *
+ * @return size_t            Size of the IPC memory handle
+ */
+typedef size_t (*opal_accelerator_base_module_get_handle_size_fn_t)(void);
+
+/**
+ * Gets an interprocess memory handle for an existing device memory allocation.
+ *
+ * @param[IN] dev_ptr        Base pointer to previously allocated device
+ *                           memory
+ * @param[OUT] handle        Pointer to user allocated mem handle to return
+ *                           the memory handle in.
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ *
+ */
+typedef int (*opal_accelerator_base_module_get_handle_fn_t)(
+    void *dev_ptr, void **handle);
+
+/**
+ * Opens an interprocess memory handle from another process and returns
+ * a device pointer usable in the local process.
+ *
+ * @param[IN] handle         IPC memory handle from another process
+ * @param[IN] dev_id         Device ID associated with the IPC memory handle
+ * @param[OUT] dev_ptr       Returned device pointer
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_open_handle_fn_t)(
+    void *handle, int dev_id, void **dev_ptr);
+
+/**
+ * Closes memory mapped with opal_accelerator_base_module_open_handle_fn_t().
+ *
+ * @param[IN] dev_ptr            IPC device pointer returned from
+ *                               opal_accelerator_base_module_open_handle_fn_t()
+ */
+typedef int (*opal_accelerator_base_module_close_handle_fn_t)(
+    void *dev_ptr);
+
+/**
+ * Page-locks the memory range specified by ptr and size
+ *
+ * @param[IN] ptr            Host pointer to memory to page-lock
+ * @param[IN] size           Size in bytes of the address range to page-lock in bytes
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_host_register_fn_t)(
```

Comment:
for habana, you need to pass device id to register a host buffer:
```
synStatus SYN_API_CALL synHostMap( const synDeviceId          deviceId,
                                   const uint64_t             size,
                                   const void*                buffer );
```

Reply:
Added device ID to each of the API's including host register.